posthog/async_migrations/setup.py

```diff
@@ -61,12 +99,13 @@ def setup_async_migrations(ignore_posthog_version: bool = False):
 
         ASYNC_MIGRATION_TO_DEPENDENCY[migration_name] = dependency
 
-        if (
-            (not ignore_posthog_version)
-            and (migration_name in unapplied_migrations)
-            and (POSTHOG_VERSION > Version(migration.posthog_max_version))
-        ):
-            raise ImproperlyConfigured(f"Migration {migration_name} is required for PostHog versions above {VERSION}.")
+    if not first_migration:
```

Comment:
Meta: We've had a lot of trouble with this code in the past and it's uncovered by tests.

When making larger behavioral changes in cases like this it's IMO important to also improve test coverage - I for on am not confident that this code actually works as intended.